### ISSUE TITLE
Obfuscator generating syntax error only on certain browsers

### ISSUE THREAD

**Issue by sgrund14** [javascript-obfuscator/javascript-obfuscator#184]
I've been using javascript-obfuscator CLI to obfuscate my code bundled with create-react-app, then deploying the obfuscated code. After publishing it online, for most browsers the code runs fine.
However, on certain browsers (like the latest firefox), the code breaks and gives a Syntax-Error: Unexpected Token. 

I run the obfuscator like so: 

`javascript-obfuscator main.6a5c981b.js --domain-lock 'www.wobc.org' --rotate-unicode-array true --compact true --control-flow-flattening false --dead-codeinjection false --debug-protection false --debug-protection-interval  false --disable-console-output true --identifier-names-generator 'hexadecimal' --log false --rename-globals false --rotate-string-array true --self-defending true --string-array true --string-array-encoding false --string-array-threshold 0.75 --unicode-escape-sequence false`

I was wondering why this problem would occur on some browsers, but not all.

**Comment by sanex3339** [comment]
Can you provide small obfuscated code that will trigger an error on firefox?

**Comment by sanex3339** [comment]
Cant reproduce on firefox 58.0.1 obfuscated Angular 2 with your setting.

**Comment by sgrund14** [comment]
`var _0x60cd58=_0x1971('0x13be')+Math[_0x1971('0x557')](),_0x32af1b=_0x1971('0xb24')==_0x14f9e8[_0x1971('0x28')][_0x1971('0x29')]?'*':_0x14f9e8[_0x1971('0x28')][_0x1971('0x29')]**+''),_0x40bfbe**=_0x3360f8[_0x1971('0x2a0')](0x0)`

The syntax error is caused by the closing paren in the bold text: Uncaught SyntaxError: Unexpected token )

**Comment by sanex3339** [comment]
can you provide small unobfuscated code that will provide this error after obfuscation?

**Comment by sgrund14** [comment]
Ok, so this is the unobfuscated code:

`if ("undefined" === typeof e && "undefined" !== typeof window && window.postMessage && window.addEventListener && !A("Presto") && (e = function() {
                        var e = document.createElement("IFRAME");
                        e.style.display = "none", e.src = "", document.documentElement.appendChild(e);
                        var t = e.contentWindow;
                        e = t.document, e.open(), e.write(""), e.close();
                        var n = "callImmediate" + Math.random(),
                            r = "file:" == t.location.protocol ? "*" : t.location.protocol + "//" + t.location.host;
                        e = p(function(e) {
                            "*" != r && e.origin != r || e.data != n || this.port1.onmessage()
                        }, this), t.addEventListener("message", e, !1), this.port1 = {}, this.port2 = {
                            postMessage: function() {
                                t.postMessage(n, r)
                            }
                        }
                    }), "undefined" !== typeof e && !A("Trident") && !A("MSIE")) {
                    var t = new e,
                        n = {},
                        r = n;
                    return t.port1.onmessage = function() {
                            if (void 0 !== n.next) {
                                n = n.next;
                                var e = n.pb;
                                n.pb = null, e()
                            }
                        },
                        function(e) {
                            r.next = {
                                pb: e
                            }, r = r.next, t.port2.postMessage(0)
                        }
                }`

This is the obfuscated code:

`if (_0x1971('0x1f') === typeof _0x140888 && _0x1971('0x1f') !== typeof window && window[_0x1971('0x1363')] && window[_0x1971('0x34')] && !_0x591ec4('Presto') && (_0x140888 = function() {
                                    var _0x140888 = document[_0x1971('0x276')](_0x1971('0x13bd'));
                                    _0x140888[_0x1971('0xa02')]['display'] = _0x1971('0x25b'), _0x140888['src'] = '', document[_0x1971('0xd6c')][_0x1971('0xa25')](_0x140888);
                                    var _0x14f9e8 = _0x140888[_0x1971('0xa13')];
                                    _0x140888 = _0x14f9e8['document'], _0x140888[_0x1971('0x9c1')](), _0x140888[_0x1971('0xa10')](''), _0x140888['close']();
                                    var _0x60cd58 = _0x1971('0x13be') + Math[_0x1971('0x557')](),
                                        _0x32af1b = _0x1971('0xb24') == _0x14f9e8[_0x1971('0x28')][_0x1971('0x29')] ? '*' : _0x14f9e8[_0x1971('0x28')][_0x1971('0x29')] + ''), _0x40bfbe = _0x3360f8[_0x1971('0x2a0')](0x0), _0x2c45cc = _0x3360f8[_0x1971('0x2a0')](0x1), _0x40bfbe + _0x2c45cc) {
                                    case _0x54f3c2:
                                        break;
                                    case _0x27028e:
                                    case _0x27541b:
                                        _0x494b45 += _0x3360f8 + _0x4daa0e['charAt'](_0x2388b6);
                                        break;
                                    default:
                                        if (_0x3360f8['charCodeAt'](_0x2fd29f - 0x1) === _0x47d324) break;
                                        _0x3101df += _0x321f65(_0x3360f8, _0x40bfbe, _0x2c45cc, _0x3360f8['charCodeAt'](0x2));
                                }
                                _0x2555c5 = 0x0, _0x20e61e = 0x0, _0x309f93 = 0x0, _0x214628 = 0x0, _0x5e99f3 = 0x0, _0x3360f8 = '', _0x5def47 = _0x4daa0e[_0x1971('0x2a0')](++_0x2388b6);
                            }`

**Comment by sanex3339** [comment]
Original and obfuscated code are different. Original code doesn't contain break statements. Can you provide the original code that will obfuscated to this code:
```js
if (_0x1971('0x1f') === typeof _0x140888 && _0x1971('0x1f') !== typeof window && window[_0x1971('0x1363')] && window[_0x1971('0x34')] && !_0x591ec4('Presto') && (_0x140888 = function() {
                                    var _0x140888 = document[_0x1971('0x276')](_0x1971('0x13bd'));
                                    _0x140888[_0x1971('0xa02')]['display'] = _0x1971('0x25b'), _0x140888['src'] = '', document[_0x1971('0xd6c')][_0x1971('0xa25')](_0x140888);
                                    var _0x14f9e8 = _0x140888[_0x1971('0xa13')];
                                    _0x140888 = _0x14f9e8['document'], _0x140888[_0x1971('0x9c1')](), _0x140888[_0x1971('0xa10')](''), _0x140888['close']();
                                    var _0x60cd58 = _0x1971('0x13be') + Math[_0x1971('0x557')](),
                                        _0x32af1b = _0x1971('0xb24') == _0x14f9e8[_0x1971('0x28')][_0x1971('0x29')] ? '*' : _0x14f9e8[_0x1971('0x28')][_0x1971('0x29')] + ''), _0x40bfbe = _0x3360f8[_0x1971('0x2a0')](0x0), _0x2c45cc = _0x3360f8[_0x1971('0x2a0')](0x1), _0x40bfbe + _0x2c45cc) {
                                    case _0x54f3c2:
                                        break;
                                    case _0x27028e:
                                    case _0x27541b:
                                        _0x494b45 += _0x3360f8 + _0x4daa0e['charAt'](_0x2388b6);
                                        break;
                                    default:
                                        if (_0x3360f8['charCodeAt'](_0x2fd29f - 0x1) === _0x47d324) break;
                                        _0x3101df += _0x321f65(_0x3360f8, _0x40bfbe, _0x2c45cc, _0x3360f8['charCodeAt'](0x2));
                                }
                                _0x2555c5 = 0x0, _0x20e61e = 0x0, _0x309f93 = 0x0, _0x214628 = 0x0, _0x5e99f3 = 0x0, _0x3360f8 = '', _0x5def47 = _0x4daa0e[_0x1971('0x2a0')](++_0x2388b6);
                            }
```

**Comment by sanex3339** [comment]
What version of obfuscator you using?
I obfuscated your source code and it runs fine:

```js
var _0x3faa = [
    'Presto',
    'createElement',
    'IFRAME',
    'none',
    'src',
    'documentElement',
    'appendChild',
    'contentWindow',
    'write',
    'close',
    'callImmediate',
    'random',
    'file:',
    'location',
    'protocol',
    'host',
    'origin',
    'data',
    'port1',
    'port2',
    'postMessage',
    'Trident',
    'next',
    'undefined',
    'addEventListener'
];
(function (_0x14e43a, _0x10faf4) {
    var _0x4f13a6 = function (_0xb7f95f) {
        while (--_0xb7f95f) {
            _0x14e43a['push'](_0x14e43a['shift']());
        }
    };
    _0x4f13a6(++_0x10faf4);
}(_0x3faa, 0x1c0));
var _0x38e6 = function (_0x48f330, _0x32d293) {
    _0x48f330 = _0x48f330 - 0x0;
    var _0x279be5 = _0x3faa[_0x48f330];
    return _0x279be5;
};
function foo() {
    if (_0x38e6('0x0') === typeof e && _0x38e6('0x0') !== typeof window && window['postMessage'] && window[_0x38e6('0x1')] && !A(_0x38e6('0x2')) && (e = function () {
            var _0x4bb496 = document[_0x38e6('0x3')](_0x38e6('0x4'));
            _0x4bb496['style']['display'] = _0x38e6('0x5'), _0x4bb496[_0x38e6('0x6')] = '', document[_0x38e6('0x7')][_0x38e6('0x8')](_0x4bb496);
            var _0x2e8e8b = _0x4bb496[_0x38e6('0x9')];
            _0x4bb496 = _0x2e8e8b['document'], _0x4bb496['open'](), _0x4bb496[_0x38e6('0xa')](''), _0x4bb496[_0x38e6('0xb')]();
            var _0x4d7e0c = _0x38e6('0xc') + Math[_0x38e6('0xd')](), _0x2bb4db = _0x38e6('0xe') == _0x2e8e8b[_0x38e6('0xf')][_0x38e6('0x10')] ? '*' : _0x2e8e8b[_0x38e6('0xf')][_0x38e6('0x10')] + '//' + _0x2e8e8b[_0x38e6('0xf')][_0x38e6('0x11')];
            _0x4bb496 = p(function (_0x299e53) {
                '*' != _0x2bb4db && _0x299e53[_0x38e6('0x12')] != _0x2bb4db || _0x299e53[_0x38e6('0x13')] != _0x4d7e0c || this[_0x38e6('0x14')]['onmessage']();
            }, this), _0x2e8e8b[_0x38e6('0x1')]('message', _0x4bb496, !0x1), this[_0x38e6('0x14')] = {}, this[_0x38e6('0x15')] = {
                'postMessage': function () {
                    _0x2e8e8b[_0x38e6('0x16')](_0x4d7e0c, _0x2bb4db);
                }
            };
        }), _0x38e6('0x0') !== typeof e && !A(_0x38e6('0x17')) && !A('MSIE')) {
        var _0x3121a3 = new e(), _0x4b8ec8 = {}, _0x50350d = _0x4b8ec8;
        return _0x3121a3[_0x38e6('0x14')]['onmessage'] = function () {
            if (void 0x0 !== _0x4b8ec8[_0x38e6('0x18')]) {
                _0x4b8ec8 = _0x4b8ec8[_0x38e6('0x18')];
                var _0x1e4f1d = _0x4b8ec8['pb'];
                _0x4b8ec8['pb'] = null, _0x1e4f1d();
            }
        }, function (_0x5603d2) {
            _0x50350d[_0x38e6('0x18')] = { 'pb': _0x5603d2 }, _0x50350d = _0x50350d[_0x38e6('0x18')], _0x3121a3['port2'][_0x38e6('0x16')](0x0);
        };
    }
}
```

**Comment by sgrund14** [comment]
I am using version 0.14.0

**Comment by sanex3339** [comment]
Can you send me a slice of your source code that will trigger this error after obfuscation? Source code above obfuscated fine for me.

**Comment by sanex3339** [comment]
Some errors in firefox can be related with
https://github.com/javascript-obfuscator/javascript-obfuscator/issues/195

**Comment by sanex3339** [comment]
Closed due inactivity. Some errors related to old FF was resolved in 0.14.3. Reopen issue if you will can attach slide of your source code that will trigger this error.
